src/Utilities/Compiler/Extensions/IOperationExtensions.cs

```diff
@@ -57,6 +57,20 @@ internal static partial class IOperationExtensions
             return typeInfo.Type as INamedTypeSymbol;
         }
 
+        /// <summary>
+        /// Gets the arguments for the current <see cref="IInvocationOperation"/> ordered by their respective parameter's <see cref="IParameterSymbol.Ordinal"/> property.
+        /// </summary>
+        public static ImmutableArray<IArgumentOperation> GetArgumentsInParameterOrder(this IInvocationOperation invocation)
```

Comment:
There is already an extension for that.

https://github.com/dotnet/roslyn-analyzers/blob/01f38d805ae9254a5615614a7dd56e792c72f1ea/src/Utilities/Compiler/Extensions/IOperationExtensions.cs#L810-L823